hook.py

```diff
@@ -19,72 +22,177 @@
 EXECUTORS = dict(command_prompt='cmd', sh='sh', powershell='psh', bash='sh')
 RE_VARIABLE = re.compile('(#{(.*?)})', re.DOTALL)
 
+DATA_DIR = 'plugins/atomic/data/'
+
 
 async def enable(services):
-    data_svc = services.get('data_svc')
+    # we only ingest data once, and save new abilities in the data/ folder of the plugin
+    if os.path.exists(DATA_DIR) and os.listdir(DATA_DIR):
+        return
```

Comment:
Let's switch this is logic to a `not` and then in that event, create a new `AtomicService` that is loaded in from the `atomic/app/atomic_svc.py`.  From them, let's move all of you logic in the hook file into the atomic service, and call those functions from the hook file.  See some of the other plugins for exactly how to do this, but in general, what it would look like is:

```Python
from plugins.atomic.app.atomic_svc import AtomicService

async def enable(services):
    if not os.path.exists(DATA_DIR) and not os.listdir(DATA_DIR):
        atomic_svc = AtomicService(services)
        atomic_svc.clone_atomic_red_team_repo()
        atomic_svc.populate_data_directory()
```

Simple example here:
https://github.com/mitre/builder/blob/master/hook.py
https://github.com/mitre/builder/blob/master/app/build_svc.py

From there, we want to reduce the cognitive complexity of each of the functions inside of the `atomic_svc` down as much as possible.

Reply:
`AtomicService` created in f6ce965